src/workflow/closure.rs

```diff
@@ -37,24 +37,25 @@ pub struct Closure {
     pub inputs: Input,
 }
 
-impl Into<Link<Closure>> for Closure {
-    fn into(self) -> Link<Closure> {
-        let mut closure_bytes = Vec::new();
-        <Closure as Into<Ipld>>::into(self).encode(DagCborCodec, &mut closure_bytes);
+impl TryFrom<Closure> for Link<Closure> {
+    type Error = anyhow::Error;
 
-        Link::new(Cid::new_v1(
+    fn try_from(closure: Closure) -> Result<Link<Closure>, Self::Error> {
+        let mut closure_bytes = Vec::new();
+        <Closure as Into<Ipld>>::into(closure).encode(DagCborCodec, &mut closure_bytes)?;
+        Ok(Link::new(Cid::new_v1(
             DagCborCodec.into(),
             Code::Sha3_256.digest(&closure_bytes),
-        ))
+        )))
     }
 }
 
-impl Into<Ipld> for Closure {
-    fn into(self) -> Ipld {
+impl From<Closure> for Ipld {
+    fn from(closure: Closure) -> Self {
         Ipld::Map(BTreeMap::from([
-            ("with".to_string(), Ipld::String(self.resource.into())),
-            ("do".to_string(), self.action.into()),
-            ("inputs".to_string(), self.inputs.into()),
+            ("with".to_string(), Ipld::String(closure.resource.into())),
+            ("do ".to_string(), closure.action.into()),
+            ("inputs".to_string(), closure.inputs.into()),
```

Comment:
Below there are many complaints from the checker. I think I disagree with all of them? The code seems reasonable, no? I guess it is relatively dense, but that's not what it's complaining about. Would it prefer these broken up across multiple variables or something?

Reply:
Well, they're gone. I have no idea. I didn't change anything 🤷‍♀️